molgenis-data-csv/src/main/java/org/molgenis/data/csv/CsvIterator.java

```diff
@@ -149,6 +151,13 @@ private Entity get()
 						next.set(name, valueList.get(colNamesMap.get(name)));
 					}
 				}
+				else if (values != null && (values.length > 1 || (values.length == 1 && values[0].length() > 0))
+						&& values.length < colNamesMap.size())
+				{
+					throw new MolgenisDataException(
+							format("Number of values (%d) doesn't match the number of headers (%d): [%s]",
+									values.length, colNamesMap.size(), stream(values).collect(joining(","))));
```

Comment:
May be a bit confusing that we're listing the parsed values, unescaped, from the line, not the actual line that got parsed.

Reply:
I considered this, but think a comma-separated value list is preferable above a message with tab separators.